libbindgen/src/ir/enum_ty.rs

```diff
@@ -49,8 +49,8 @@ impl Enum {
         }
 
         let declaration = ty.declaration().canonical();
-        let repr = Item::from_ty(&declaration.enum_type(), None, None, ctx)
-            .ok();
+        let et = &declaration.enum_type().expect("This should be an enum since we checked above!");
```

Comment:
CI is failing because the type can indeed be invalid (see `libbindgen/tests/headers/func_ptr_in_struct.h`), so you'll need to change the code to preserve the old behavior, something like:

```rust
let repr = declaration.enum_type()
    .and_then(|ty| Item::from_ty(&ty, None, None, ctx))
    .ok();
```